Stacks: update to 1.46 + summary HTML report

abretaud opened the pull request:
Hi all,
Added calls to stacks_summary (by @mariabernard) to produce beautiful HTML report from stacks output.
@mariabernard can you look at this and/or test it?
I didn't add the commands for genotypes and populations as we don't currently save the log file. Is it required to produce good reports?

mariabernard commented:
the log file is only required when cstacks is launched, so for cstacks, denovo_map and ref_map.

abretaud commented:
Ok, I finished updating the wrappers to 1.46, added missing bits of stacks_summary, fixed a few bugs and made a bit of cleanup.
Tests are (almost) all green when testing with a local install of stacks.

There are 2 problems left:
-a segfault in stacks_populations test (reported there: https://groups.google.com/forum/#!topic/stacks-users/uqm1peUEod4)
-the stacks bioconda recipe is not yet available as it requires on a newer version of gcc (https://github.com/bioconda/bioconda-recipes/pull/3987)

So it cannot be merged yet, but it can be reviewed/tested!
ping @lecorguille (in case you have not seen this PR)

abretaud commented:
We now have a workaround for the segfault :)

bgruening commented:
@abretaud where is it? :)

abretaud commented:
There: https://github.com/galaxyproject/tools-iuc/pull/987/commits/b6439930dcba993ee32f1f17be9aef848adc0634#diff-cc5f39592bae6848b3049fdbbb110555L367
I removed the (not needed) option that was provoking the segfault, and the segfault itself should be fixed in next stacks version

bgruening commented:
@abretaud great! Can they also relax the requirement for a newer GCC ;)

abretaud commented:
I haven't asked (yet!)
No news on the possible newer gcc in conda?

bgruening commented:
Not yet, no :(

abretaud commented:
The conda package is ok now :)
But I get [this error](https://travis-ci.org/galaxyproject/tools-iuc/jobs/223082420#L481) now :(
Any idea?

bgruening commented:
There seems to be an error in your shed.yaml file.

abretaud commented:
Holy cow! It's green!

bgruening commented:
Yeah!!!!

bgruening commented on tools/stacks/stacks_denovomap.xml:
```diff
@@ -6,54 +6,47 @@
+                #if $input_parent.is_of_type('fastqsanger')
+                    #set $data_path = $clean_ext($input_parent.element_identifier) + ".fq"
+                #else if $input_parent.is_of_type('fastqsanger.gz')
+                    #set $data_path = $clean_ext($input_parent.element_identifier) + ".fq.gz"
```
Is there support for bz2 files?

abretaud replied:
No unfortunately, only gz for now

bgruening commented on tools/stacks/stacks_ustacks.xml:
```diff
@@ -75,28 +74,38 @@
-             > ustacks.log 2>&1
+            2>&1 | tee ustacks.log
+
+            ## If input is in gz format, stacks will output gzipped files (no option to control this)
```
Can you not write something like `if is_of_type(foobar.gz): gunzip bar/*`

abretaud replied:
Done for ustacks! For denovo_map it's a little trickier, we feed him multiple files and I think it can be a mix of fq and fq.gz. I don't know what stacks will do in this case...

abretaud replied:
Self-answering: it will produce a mix of zipped/non-zipped tsv files

bgruening commented:
This is really a nice work! I added two small comments.

abretaud commented:
Hey,
@bgruening is it ok for you now?
Anyone else willing to review it or do we merge it?

bgruening commented:
Sure! Thanks!

abretaud commented:
Wunderbar! Thank you!

lecorguille commented:
Thanks a bunch, like they use to say!
